### ISSUE TITLE
add support for links

### ISSUE THREAD

**Issue by ItsVeryWindy** [hoptical/nodegraph-api-plugin#20]
**Is your feature request related to a problem? Please describe.**
I was trying to figure out why the context menu wasn't being displayed, then I came across this issue https://github.com/hoptical/nodegraph-api-plugin/issues/17. Doing some more digging I then came across this other issue https://github.com/grafana/grafana/issues/33358. So I thought great, I'd probably want to return some links anyway, lets see if I add some links if it fixes it. But from what I can see links don't appear to be supported by this plugin.

**Describe the solution you'd like**
I'm not too particular on the implementation, but it will most likely have to be part of the payload returned by the endpoint. At least for my purposes, I'm only interested in being able to add "external" links.

**Additional context**
These can be seen and mentioned on the page for the visualisation https://grafana.com/docs/grafana/latest/visualizations/node-graph/, but I didn't have much luck finding documentation on implementing them.

This I believe describes the interface for links. The more interesting properties are `url`, `title`, `targetBlank`, which are hopefully self explanatory.
https://grafana.com/docs/grafana/latest/packages_api/data/datalink/#datalink-interface

Links can be added as an array as part of the config, around this part of the code.
https://github.com/hoptical/nodegraph-api-plugin/blob/eaf0cc88c375024c67912afa5b6986db2d301c2d/src/datasource.ts#L58
eg.
```javascript
outputField.config.links = [{
url: 'http://example.com',
title: 'Example'
}];
```
I did play around with it a bit hard coding stuff in there, and I was able to get the context menu and links to display.

**Comment by lmangani** [comment]
I'm after the same challenge, but targeting internal links to other datasources. Thanks for opening this issue!
 @ItsVeryWindy could you check if the proposed PR #21 might work for your use case?

**Comment by lmangani** [comment]
Good point... perhaps we might iterate like the details do - ie: `link__name__property`?

**Comment by perotta** [comment]
I created a more generic solution if you are interested on it PR #23

**Comment by hoptical** [comment]
Dear contributors,

Thank you for your contribution and interest in this project.
I’m sorry for the late response.
This repository is now archived and no longer maintained.
We recommend using the [Infinity plugin](https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/) as a more flexible and actively maintained alternative for connecting REST APIs to Grafana.
